src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Suppression
+{
+    internal partial class AbstractSuppressionCodeFixProvider
+    {
+        /// <summary>
+        /// Helper methods for pragma suppression add/remove batch fixers.
+        /// </summary>
+        private static class PragmaBatchFixHelpers
+        {
+            public static CodeAction CreateBatchPragmaFix(
+                AbstractSuppressionCodeFixProvider suppressionFixProvider,
+                Document document,
+                ImmutableArray<IPragmaBasedCodeAction> pragmaActions,
+                ImmutableArray<Diagnostic> pragmaDiagnostics,
+                FixAllContext fixAllContext)
+            {
+                // This is a temporary generated code action, which doesn't need telemetry, hence suppressing RS0005.
+#pragma warning disable RS0005 // Do not use generic CodeAction.Create to create CodeAction
```

Comment:
so for this kind of "suppression", how it is supposed to be used with this new feature we are adding. this is probably not considered as base line suppression, so is there a way to see the comment from error list so that when a user do unsuppress, they see this suppression is not part of base line?